pkg/scheduler/framework/plugins/default_registry.go

```diff
@@ -105,6 +105,15 @@ func NewDefaultConfigProducerRegistry() *ConfigProducerRegistry {
 		PredicateToConfigProducer: make(map[string]ConfigProducer),
 		PriorityToConfigProducer:  make(map[string]ConfigProducer),
 	}
+	registry.RegisterPredicate(predicates.GeneralPred,
+		func(_ ConfigProducerArgs) (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			// GeneralPredicate is a combination of predicates.
```

Comment:
Is it cleaner to call `RegisterPredicate` each time for each predicate instead of combining them in one "general predicate"? Or what's special about "GeneralPred" compared to others?

Reply:
Unfortunately "GeneralPredicate" is registered as a "composite" [standalone predicate](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithmprovider/defaults/register_predicates.go#L102). By composite I mean that its implementation basically calls a group of other predicates. I think it was a mistake making it a predicate on of its own, but we are stuck with it since it is now a value that we deem valid in the schedule config file.

Reply:
ack.

Reply:
As @ahg-g said, we were not doing well in the "GeneralPredicate" thing - it shouldn't be considered as, or registered as a serious Predicate, instead, it should serve as a helper to invoke a series of **real** Predicate, and called by consumers like controller manager and kubelet.

To make our code readable and the semantics of Predicates/FilterPlugins clear, I hope we can get rid of the "GeneralPredicate" thing in 2 aspects:

- Rename it to be not related with "Predicate" - just serve as a "composite" to invoke other Predicate/Plugin
- Discard the support to expose it as a regular Predicate in Policy API.

WDYT?

Reply:
I am happy to rename/discard it, but it is part of the Policy v1 API. If someone has a custom policy configuration that included "GeneralPredicate" as one of the priorities, their scheduler will fail to start. 

However, if we do remove it as a predicate in the policy API, it is easy for users to migrate, they just need to modify their policy file to include the corresponding predicates, and it is backward compatible (i.e., specifying the list of corresponding predicate will work for new and older versions of the scheduler). I am fine with this if we get an API review approval for it, but that should be in a separate PR.

Reply:
> if we do remove it as a predicate in the policy API, it is easy for users to migrate, they just need to modify their policy file to include the corresponding predicates, and it is backward compatible

Yup, it sounds good to me.